ClangSharpPInvokeGenerator/CursorWriter.cs

```diff
@@ -932,6 +981,47 @@ private bool BeginHandleUnaryOperator(CXCursor cursor, CXCursor parent)
             }
         }
 
+        private bool BeginHandleUnexposedAttr(CXCursor cursor, CXCursor parent)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_UnexposedAttr);
+
+            if (_attachedData.TryGetValue(parent, out var data) && (data is AttachedFunctionDeclData functionDeclData))
+            {
+                var translationUnit = cursor.TranslationUnit;
+                translationUnit.Tokenize(cursor.Extent, out CXToken[] tokens);
+
+                if (tokens is null)
+                {
+                    var token = translationUnit.GetToken(cursor.Location);
+                    Debug.Assert(!token.Equals(default(CXToken)));
+                    tokens = new CXToken[] { token };
+                }
+                Debug.Assert(tokens.Length == 1);
+
+                if (tokens[0].Kind != CXTokenKind.CXToken_Identifier)
+                {
+                    Unhandled(cursor, parent);
+                    return false;
+                }
+
+                var identifier = tokens[0].GetSpelling(translationUnit).ToString();
+
+                if (identifier != "deprecated")
+                {
+                    Unhandled(cursor, parent);
+                    return false;
+                }
+
+                functionDeclData.IsDeprecated = true;
```

Comment:
ClangSharp currently has just one of these and it would be nice to surface correctly. Maybe via the `[Obsolete]` attribute.